include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,44 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  ///@{
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void append(span<const TensorListShape<sample_ndim>> tlss) {
+    int new_samples = 0;
+    for (const auto & tls : tlss) {
+      new_samples += tls.num_samples();
+    }
+    if (new_samples == 0) return;
+    auto reference_dim_value = empty() ? tlss[0].sample_dim() : sample_dim();
```

Comment:
I don't think being empty is enough to take the sample_dim from the argument. If `this` already has some dimension set, it would produce an explicit error (as explained in comment below), and this is inconsistent behaviour. I think shape manipulation should be made safe if possible.